File: src/main/vagrant/oraclexe/scripts/create-default-db-objects.py
#!/usr/bin/python
import jaydebeapi
import sys
import time

con = None
semconexao = True
MAX_TENTATIVA = 10
tentativa = 1

while (tentativa <= MAX_TENTATIVA and semconexao):
  print "Tentando criar objetos de banco de dados... ", tentativa, " de ", MAX_TENTATIVA
  sys.stdout.flush()
  try:
    con = jaydebeapi.connect('oracle.jdbc.driver.OracleDriver',
                              'jdbc:oracle:thin:@localhost:1521:xe', [ 'system', 'oracle'],
                              r'/vagrant/scripts/ojdbc7-12.1.0.jar')
    semconexao = False
  except Exception, e:
    print e
    time.sleep(30)
  tentativa = tentativa + 1
  sys.stdout.flush()

if semconexao:
  print "Nao foi possivel os objetos de banco de dados banco de dados."
  sys.exit(1)

curs = con.cursor()
curs.execute("CREATE USER PTOLOMEU IDENTIFIED BY PTOLOMEU")
curs.execute("grant connect to PTOLOMEU")
curs.execute("grant all privileges to PTOLOMEU")
curs.close()
con.close()

print "Objetos de BD foram criados com sucesso!"

File: src/main/vagrant/oraclexe/scripts/src/main/java/oracle/CreateDefaultDbObjects.java
package oracle;
import java.util.concurrent.TimeUnit;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.sql.Statement;

public class CreateDefaultDbObjects {

    public static void main(String[] args) throws ClassNotFoundException, InterruptedException {
        String oracle_xe_host = System.getenv("ORACLE_XE_HOST");
        if (oracle_xe_host == null) {
            oracle_xe_host = "localhost";
        }
        String jdbc_url = String.format("jdbc:oracle:thin:@%s:1521:xe", oracle_xe_host);
        System.out.println("JDBC URL: " + jdbc_url);


        Class.forName("oracle.jdbc.driver.OracleDriver");
        Connection connection = null;
        int MAX_TENTATIVA = 20;
        int tentativa = 1;
        boolean sem_conexao = true;

        while (tentativa <= MAX_TENTATIVA && sem_conexao) {
            System.out.println("Tentando criar objetos de banco de dados... " + tentativa + " de " + MAX_TENTATIVA);
            try {
                connection = DriverManager.getConnection(jdbc_url,"system","oracle");
                Statement statement = connection.createStatement();
                statement.execute("CREATE USER sisas IDENTIFIED BY sisas");
                statement.execute("grant connect to sisas");
                statement.execute("grant all privileges to sisas");
                System.out.println("Objetos de BD foram criados com sucesso!");
                statement.close();
                connection.close();
                sem_conexao = false;
            } catch (SQLException e) {
                System.out.println(e.toString());
                TimeUnit.SECONDS.sleep(20);
            }
            tentativa++;
        }

        if (sem_conexao) {
            System.out.println("Nao foi possivel conectar ao banco de dados.");
            System.exit(1);
        }
    }

}
